s390x: test_python2/test_python2_unicode_literals can't assign to () INTERNAL ERROR

During the build of 19.10b0 in Fedora, the following test failure occurs on s390x (Big Endian) architecture:

```
======================================================================
FAIL: test_python2 (tests.test_black.BlackTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/black-19.10b0/black.py", line 3754, in assert_equivalent
    src_ast = parse_ast(src)
  File "/builddir/build/BUILD/black-19.10b0/black.py", line 3686, in parse_ast
    return ast27.parse(src)
  File "/usr/lib64/python3.8/site-packages/typed_ast/ast27.py", line 50, in parse
    return _ast27.parse(source, filename, mode)
  File "<unknown>", line 10
SyntaxError: can't assign to ()
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/mock.py", line 1342, in patched
    return func(*newargs, **newkeywargs)
  File "/builddir/build/BUILD/black-19.10b0/tests/test_black.py", line 543, in test_python2
    black.assert_equivalent(source, actual)
  File "/builddir/build/BUILD/black-19.10b0/black.py", line 3756, in assert_equivalent
    raise AssertionError(
AssertionError: cannot use --safe with this file; failed to parse source file.  AST error message: can't assign to () (<unknown>, line 10)
======================================================================
FAIL: test_python2_unicode_literals (tests.test_black.BlackTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/mock.py", line 1342, in patched
    return func(*newargs, **newkeywargs)
  File "/builddir/build/BUILD/black-19.10b0/tests/test_black.py", line 560, in test_python2_unicode_literals
    black.assert_equivalent(source, actual)
  File "/builddir/build/BUILD/black-19.10b0/black.py", line 3775, in assert_equivalent
    raise AssertionError(
AssertionError: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: 
--- src
+++ dst
@@ -1,4 +1,70 @@
 Module(
   body=
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            '_unicode_literals',  # str
+          name=
+            'unicode_literals',  # str
+        )  # /alias
+    )  # /ImportFrom
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            None,  # NoneType
+          name=
+            'absolute_import',  # str
+        )  # /alias
+    )  # /ImportFrom
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            'lol',  # str
+          name=
+            'print_function',  # str
+        )  # /alias
+        alias(
+          asname=
+            None,  # NoneType
+          name=
+            'with_function',  # str
+        )  # /alias
+    )  # /ImportFrom
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
   type_ignores=
 )  # /Module
----------------------------------------------------------------------
Ran 119 tests in 18.012s
FAILED (failures=2)
```

**To Reproduce**, run the test suite on s390x. 

Here is the build log with all the commands: [build.log](https://github.com/psf/black/files/3782557/build.log)

Here is the root log with all the package versions: [root.log](https://github.com/psf/black/files/3782561/root.log)


**Expected behavior**
Test succeed an all architectures.

**Environment:**

- Version: 19.10b0
- OS and Python version: Linux, Fedora 32 on s390x, Python 3.8.0

**Does this bug also happen on master?** yes, on 6bedb5c58a7d8c25aa9509f8217bc24e9797e90d

**Additional context** The problem does not happen on the same build system with armv7hl or ppc64le.

On master 6bedb5c58a7d8c25aa9509f8217bc24e9797e90d:

```
======================================================================
FAIL: test_python2 (tests.test_black.BlackTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/black-master/black.py", line 3754, in assert_equivalent
    src_ast = parse_ast(src)
  File "/builddir/build/BUILD/black-master/black.py", line 3686, in parse_ast
    return ast27.parse(src)
  File "/usr/lib64/python3.8/site-packages/typed_ast/ast27.py", line 50, in parse
    return _ast27.parse(source, filename, mode)
  File "<unknown>", line 10
SyntaxError: can't assign to ()
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/mock.py", line 1342, in patched
    return func(*newargs, **newkeywargs)
  File "/builddir/build/BUILD/black-master/tests/test_black.py", line 543, in test_python2
    black.assert_equivalent(source, actual)
  File "/builddir/build/BUILD/black-master/black.py", line 3756, in assert_equivalent
    raise AssertionError(
AssertionError: cannot use --safe with this file; failed to parse source file.  AST error message: can't assign to () (<unknown>, line 10)
======================================================================
FAIL: test_python2_unicode_literals (tests.test_black.BlackTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.8/unittest/mock.py", line 1342, in patched
    return func(*newargs, **newkeywargs)
  File "/builddir/build/BUILD/black-master/tests/test_black.py", line 560, in test_python2_unicode_literals
    black.assert_equivalent(source, actual)
  File "/builddir/build/BUILD/black-master/black.py", line 3775, in assert_equivalent
    raise AssertionError(
AssertionError: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: 
--- src
+++ dst
@@ -1,4 +1,70 @@
 Module(
   body=
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            '_unicode_literals',  # str
+          name=
+            'unicode_literals',  # str
+        )  # /alias
+    )  # /ImportFrom
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            None,  # NoneType
+          name=
+            'absolute_import',  # str
+        )  # /alias
+    )  # /ImportFrom
+    ImportFrom(
+      level=
+        0,  # int
+      module=
+        '__future__',  # str
+      names=
+        alias(
+          asname=
+            'lol',  # str
+          name=
+            'print_function',  # str
+        )  # /alias
+        alias(
+          asname=
+            None,  # NoneType
+          name=
+            'with_function',  # str
+        )  # /alias
+    )  # /ImportFrom
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
+    Expr(
+      value=
+        Constant(
+          value=
+            'hello',  # str
+        )  # /Constant
+    )  # /Expr
   type_ignores=
 )  # /Module
----------------------------------------------------------------------
Ran 119 tests in 18.123s
FAILED (failures=2)
```

--- a/Pipfile
+++ b/Pipfile
@@ -28,7 +28,7 @@ mypy_extensions = ">=0.4.3"
 pathspec = ">=0.6"
 regex = ">=2020.1.8"
 toml = ">=0.10.1"
-typed-ast = "==1.4.1"
+typed-ast = "==1.4.2"
 typing_extensions = ">=3.7.4"
 black = {editable = true,extras = ["d"],path = "."}
 dataclasses = {"python_version <" = "3.7","version >" = "0.6"}

--- a/setup.py
+++ b/setup.py
@@ -71,7 +71,7 @@ setup(
         "click>=7.1.2",
         "appdirs",
         "toml>=0.10.1",
-        "typed-ast>=1.4.0",
+        "typed-ast>=1.4.2",
         "regex>=2020.1.8",
         "pathspec>=0.6, <1",
         "dataclasses>=0.6; python_version < '3.7'",

